core/src/test/java/org/bitcoinj/core/ECKeyTest.java

```diff
@@ -72,15 +70,18 @@ public void setUp() throws Exception {
     public void sValue() throws Exception {
         // Check that we never generate an S value that is larger than half the curve order. This avoids a malleability
         // issue that can allow someone to change a transaction [hash] without invalidating the signature.
-        final int ITERATIONS = 10;
-        ListeningExecutorService executor = MoreExecutors.listeningDecorator(Executors.newFixedThreadPool(ITERATIONS));
-        List<ListenableFuture<ECKey.ECDSASignature>> sigFutures = new ArrayList<>();
+        final byte ITERATIONS = 10;
         final ECKey key = new ECKey();
-        for (byte i = 0; i < ITERATIONS; i++) {
-            final Sha256Hash hash = Sha256Hash.of(new byte[]{i});
-            sigFutures.add(executor.submit(() -> key.sign(hash)));
-        }
-        List<ECKey.ECDSASignature> sigs = Futures.allAsList(sigFutures).get();
+
+        Function<Byte, ECKey.ECDSASignature> signer = b -> key.sign(Sha256Hash.of(new byte[]{b}));
+        Function<Byte, CompletableFuture<ECKey.ECDSASignature>> asyncSigner = b -> CompletableFuture.supplyAsync(() -> signer.apply(b));
+
+        List<CompletableFuture<ECKey.ECDSASignature>> sigFutures = IntStream.range(0, ITERATIONS - 1)
+                .mapToObj(i -> (byte) i)
+                .map(asyncSigner)
```

Comment:
Seems to me that instead if signing `hash(i)` we're now signing just `i`. Is that intentional? It probably doesn't matter much though.

Reply:
The `.map(asyncSigner)` step calls the `asyncSinger` function, which calls the `signer` function which does the hash.

Reply:
Ah, I missed the Sha256Hash in the function.